components/features/Dashboard/Applications/InteractiveQuiz/AssignmentTimer/AssignmentTimer.tsx

```diff
@@ -0,0 +1,62 @@
+import React, { useState, useEffect, useRef, useMemo } from "react";
+import styles from "./AssignmentTimer.module.css";
+
+interface Props extends React.PropsWithChildren {
+  durationInMinutes: number;
+  onTimerEnd?: () => void;
+}
+
+const timeFormatter = (totalSeconds: number) => {
```

Comment:
Hello, rashad! Maybe in the future, if possible, we can refactor the time formatter into its own utils function, so we can use it somewhere else in our codebase. 😁

Reply:
Hi Alex good catch, yes I do plan on extracting this to a separate util function in the future. Thanks for the feedback :D